apps/04TowersOfHanoi.js

```diff
@@ -18,31 +18,44 @@ function printStacks() {
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
-
-function movePiece() {
+//Popping and pushing the startstack and endstack so the pieces will move.
+function movePiece(startStack, endStack) {
   // Your code here
+  var begin = stacks[startStack].pop();
+  stacks[endStack].push(begin);
 
 }
 
-function isLegal() {
-  // Your code here
-
+//Making sure the moves are legal moves.
+function isLegal(startStack, endStack) {
+  var start = stacks[startStack].slice(-1)
+  var end = stacks[endStack].slice(-1)
+  if (stacks[endStack].length === 0) {
+    return true
+  }
+  return start < end
 }
-
+//Checking to see if it's a valid move and seeing if there is a win.
 function checkForWin() {
-  // Your code here
-
+  console.log("You Win!");
+  return stacks.b.length === 4 || stacks.c.length === 4
 }
-
+//If not a valid move it says that's not a valid move.
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
-
+  if (isLegal(startStack, endStack)) {
+    movePiece(startStack, endStack)
+    getPrompt();
```

Comment:
We actually don't need these here, since `getPrompt` is called automatically at the end of `getPrompt`